### DISCUSSION TITLE
(PVE Perl Style cont.) Prefer newline after opening brace if the container is the last parameter

### DISCUSSION THREAD

**Discussion by cheese** [perltidy/perltidy#96] [Q&A]
I added `--maximum-fields-per-table=1` to my perltidyrc:
```
--indent-columns=4
--entab-leading-whitespace=8
--maximum-fields-per-table=1
--maximum-line-length=100
--weld-nested-containers
--paren-tightness=2
--square-bracket-tightness=0
--brace-tightness=0
--cuddled-else
--opening-token-right
--opening-brace-always-on-right
--stack-all-containers
--stack-closing-block-brace
--tight-secret-operators

--no-outdent-long-lines
--continuation-indentation=4
--novalign
--noblanks-before-comments
--character-encoding=utf8
--warning-output
--output-line-ending=unix
--space-for-semicolon
--minimum-space-to-comment=1
--ignore-side-comment-lengths
```

What I expect:
```
get_standard_option('pve-storage-id', {
    description => "Only list status for  specified storage",
    optional => 1,
    completion => \&PVE::Storage::complete_storage_enabled,
}),
```

What I got from perltidy:
```
get_standard_option(
    'pve-storage-id', {
        description => "Only list status for  specified storage",
        optional => 1,
        completion => \&PVE::Storage::complete_storage_enabled,
    }
    ),
```

Note:
1. If the last parameter is a container and the point of the opening brace does not pass the line limit, then new line added after the opening brace instead of the opening of parenthesis.
2. All ending symbols are stacked no matter they are end of parameters or end of containers.
3. The stacked ending symbols are placed at the start of line

**Comment by shancock9** [comment]
I don't see a way to duplicate this.  As a general rule, if perltidy "breaks open" a container into multiple lines, as in the '{' here, then there will automatically be breaks placed after any outer ('parent') opening containers, such as after the '(' here.  That rule is essential for displaying complex structures.  The main exception is when opening and closing container tokens can be welded (-wn).  So in this case the line break after the paren cannot be prevented.

If the snippet had been a complete statement ending in a ';' rather than a ',' then the two closing tokens would be together on the same line.  But they would still have to be indented in this particular case because the closing brace is not allowed to be indented less than the line with the corresponding opening brace.
